app/lib/katello/validators/container_image_name_validator.rb

```diff
@@ -8,7 +8,7 @@ def validate_each(record, attribute, value)
       end
 
       def self.validate_name(name)
-        if name.empty? || name.length > 255 || !/\A([a-z0-9]+[a-z0-9\-_.]*)+(\/[a-z0-9]+[a-z0-9\-_.]*)*\z/.match?(name)
+        if name.empty? || name.length > 255 || !/\A([a-z0-9]+([\-_.]*[a-z0-9]+)*)+(\/[a-z0-9]+([\-_.]*[a-z0-9]+)*)*\z/.match?(name)
```

Comment:
Here is the official regex from the OCI spec:
```
[a-z0-9]+([._-][a-z0-9]+)*(/[a-z0-9]+([._-][a-z0-9]+)*)*
```

https://specs.opencontainers.org/distribution-spec/?v=v1.0.0#DISTRIBUTION-SPEC-27

Reply:
There seems to be a small difference to how pulp_container defines it, they do a `split('/')` first and check each element separately:
```
[a-z0-9]+(?:(?:[._]|__|[-]*)[a-z0-9])*
```
https://github.com/pulp/pulp_container/blob/main/pulp_container/app/serializers.py#L39

The official regex would suggest that `.`, `-`, and `_` must be followed by alphanumerical character so `a__b` or `c---d` would not be allowed. While pulp seems to explicitly allow them :dizzy_face: 
If we choose the official Regexp we should make sure that pulp uses that one as well.

Reply:
Yeah, I mostly posted this in case it was interesting, we don't need to open this can of worms yet :)

Reply:
Nice catch though, I can ask them why their regex is different from the OCI one. Maybe the OCI spec changed at some point.